README.rst

```diff
@@ -1,77 +1,128 @@
+
+For a more detailed explanation of installing plug-ins, refer to the `Curity developer guide <https://developer.curity.io/docs/latest/developer-guide/plugins/index.html#plugin-installation>`_.
+
+Creating an App in Github
+~~~~~~~~~~~~~~~~~~~~~~
+
+As `described in the Github documentation <https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app>`_, You can `create and register <https://github.com/settings/applications/new>`_ an OAuth App under your personal account or under any organization you have administrative access to.
+
+    .. figure:: docs/images/create-github-app1.png
+            :name: new-github-app
+            :align: center
+            :width: 500px
+
+Creating a new Github application
+
+    .. figure:: docs/images/create-github-app2.png
+            :name: create-github-app
+            :align: center
+            :width: 500px
+
+Selecting to authenticate users in the new app
+
+Then, give the app a name, e.g., ``Curity-Enterprise-Integration-App``.
+
+When you view the app's configuration after creating it, you'll find the ``Client ID`` and ``Client Secret``. These will be needed later when configuring the plug-in in Curity.
+
+Github will also display the Authorization callback URL in the new app's configuration. This needs to match the yet-to-be-created Github authenticator instance in Curity. The default will not work, and, if used, will result in an error. This should be updated to some URL that follows the pattern ``$baseUrl/$authenticationEndpointPath/$githubAuthnticatorId/callback``, where each of these URI components has the following meaning:
+
+============================== =========================================================================================
+URI Component                  Meaning
+------------------------------ -----------------------------------------------------------------------------------------
+``baseUrl``                    The base URL of the server (defined on the ``System --> General`` page of the
+                               admin GUI). If this value is not set, then the server scheme, name, and port should be
+                               used (e.g., ``https://localhost:8443``).
+``authenticaitonEndpointPath`` The path of the authentication endpoint. In the admin GUI, this is located in the
+                               authentication profile's ``Endpoints`` tab for the endpoint that has the type
+                               ``auth-authentication``.
+``githubAuthenticatorId``         This is the name given to the Github authenticator when defining it (e.g., ``github1``).
+============================== =========================================================================================
+
+Once the redirect URI is updated, the app is ready to be used from Curity.
+
+Creating a Github Authenticator in Curity
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
the `~` should go all the way to the end of the text `Curity`. I get warnings about this in Intellij. If you don't, you may want to update your Intellij settings.

Reply:
Intellij does show warning on some headings, but was not showing for ``~`` one's